### DISCUSSION TITLE
Extra Hot Pixels

### DISCUSSION THREAD

**Discussion by hollasch** [RayTracing/raytracing.github.io#926] [General]
Great discussion on the issue of extra hot (white) pixels when rendering: #660

**Comment by Zi7ar21** [comment]
Does "hot"/white pixels refer to fireflies or undefined floating point behavior? The solution to fireflies is radiance clamping or adjusting sampling. Undefined floating point behavior is when a sample somehow ends up as a NaN or Inf (e.g. divided by 0 somewhere) and turns the pixel black/white/something when it's accumulated.

**Reply by trevordblack** [reply]
"hot" pixels is a bit of a catch-all. It's any pixel that is perceived as being anomalously bright when compared to its neighbors.

It usually refers to fireflies, but fireflies are specifically pixels that are perceived as being anomalously bright due to ray termination at a light.

There may be other sources of hot pixels that are not due to ray termination at a light, that would depend on your renderer (and what you consider a light).

Most renderers (image magick, FastStone) will render NaNs as black, not sure how universal that is, but I've never seen a NaN render as white. NaNs have the property that they are viral. Most operations involving NaNs will create a NaN as the output
